### ISSUE TITLE
selinux/podman issue for rootless pod: error while loading shared libraries: /lib64/libc.so.6: cannot apply  ....

### ISSUE THREAD

**Issue by groovyman** [containers/podman#22384]
### Issue Description

I set up a fresh fedora f39 server in order to get some rootless podman on my new intel server. The server installation went on a small ssd, where the technical users, that will run the rootless-container  services using podman are located on a raid-mount. The technical users have been set up using the cockpit-ui.  They techuser are located under '/mnt/raider/home2/techuser01'.

# The problem looks like this:
```
$ podman  run -it --rm registry.fedoraproject.org/fedora:latest
/bin/bash: error while loading shared libraries: /lib64/libc.so.6: cannot apply additional memory protection after relocation: Permission denied
```
# Here is a list of the installed components ( Fedora-Server F39 is updated frequently)
- podman-4.9.4-1.fc39.x86_64
- podman info gives me:  (see following text field

# What i tried in the meantime
@rhatdan wrote, that all selinux related problems with shared libs are configuration problems. Yes i disabled selinux and the problem disappeard. The i tried the proposed fix/workarounds  described in 
- #3234  using the semanage and restorecon [moved container storage ... no it was not but i tried](https://github.com/containers/podman/issues/3234#issuecomment-497541854)
- #10817 some removals of my '...containers/*' directory and a `podman system reset`  

I have no idea where to look now, i am done. The only thing i am thinking of is a betes-noir i configured with the '/mnt/raider'  mount.

### Steps to reproduce the issue

Steps to reproduce the issue
1. create a technical user on a mounted raid drive
2. with this tech-user issue the call 'podman  run -it --rm registry.fedoraproject.org/fedora:latest'
3.

### Describe the results you received

/bin/bash: error while loading shared libraries: /lib64/libc.so.6: cannot apply additional memory protection after relocation: Permission denied

### Describe the results you expected

a running shell or test

### podman info output

```yaml
$ podman info
host:
  arch: amd64
  buildahVersion: 1.33.7
  cgroupControllers:
  - cpu
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.10-1.fc39.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.10, commit: '
  cpuUtilization:
    idlePercent: 99.68
    systemPercent: 0.18
    userPercent: 0.14
  cpus: 4
  databaseBackend: sqlite
  distribution:
    distribution: fedora
    variant: server
    version: "39"
  eventLogger: journald
  freeLocks: 2048
  hostname: chasmash
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 2131
      size: 1
    - container_id: 1
      host_id: 720896
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 2131
      size: 1
    - container_id: 1
      host_id: 720896
      size: 65536
  kernel: 6.8.5-201.fc39.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 31545425920
  memTotal: 33467539456
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.10.0-1.fc39.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.10.0
    package: netavark-1.10.3-1.fc39.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.10.3
  ociRuntime:
    name: crun
    package: crun-1.14.4-1.fc39.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.14.4
      commit: a220ca661ce078f2c37b38c92e66cf66c012d9c1
      rundir: /run/user/2131/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt-0^20240326.g4988e2b-1.fc39.x86_64
    version: |
      pasta 0^20240326.g4988e2b-1.fc39.x86_64
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: false
    path: /run/user/2131/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.2-1.fc39.x86_64
    version: |-
      slirp4netns version 1.2.2
      commit: 0ee2d87523e906518d34a6b423271e4826f71faf
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.3
  swapFree: 8589930496
  swapTotal: 8589930496
  uptime: 13h 14m 27.00s (Approximately 0.54 days)
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /mnt/raider/home2/techlxoffice/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /mnt/raider/home2/techlxoffice/.local/share/containers/storage
  graphRootAllocated: 11965456449536
  graphRootUsed: 83919908864
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Supports shifting: "false"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 1
  runRoot: /run/user/2131/containers
  transientStore: false
  volumePath: /mnt/raider/home2/techlxoffice/.local/share/containers/storage/volumes
version:
  APIVersion: 4.9.4
  Built: 1711445992
  BuiltTime: Tue Mar 26 10:39:52 2024
  GitCommit: ""
  GoVersion: go1.21.8
  Os: linux
  OsArch: linux/amd64
  Version: 4.9.4
```

### Podman in a container

Yes

### Privileged Or Rootless

Rootless

### Upstream Latest Release

Yes

### Additional environment details

Additional environment details

### Additional information

none

**Comment by rhatdan** [comment]
This is an SELinux issue. 

You used an alternate directory for container content, not in a standard directory.

https://github.com/containers/storage/blob/main/docs/containers-storage.conf.5.md#storage-table

Did you fix the SELinux labels to work correctly?

You could also submit the AVC messages.

sudo ausearch -m avc -ts recent.

**Comment by groovyman** [comment]
Salut 

For the first: i did not changed directory of the container content. I simply used that one, that was given by the created account. But ..  it must have been my fault. I created two accounts:

- techuser01 that resides of the boot drive (ssd)  
- techuser02 that resides on the mounted raid

For both cases i tested the above shown command `podman  run -it --rm registry.fedoraproject.org/fedora:latest`  and both worked! So it is my fault an i will close that issue. I guess that the used techaccount was botched by myself. 
So i will drop the techuser and recreate it.

Anyway i will have to undo the system modification for that user, that may disturb future actions taken place on a newly created account, that uses the same account name. Am i right ?
